visual_diff.py

```diff
@@ -49,6 +47,10 @@ def get_tokens(filename):
     matrix[i, :] = (tokens_b == value)
 
 if args.gui:
+    # Usually it's a bad idea to import modules way down here, but we do this
+    # so that people who aren't able to install the TK bindings for PIL are
+    # still able to use the non-GUI version of this script.
+    import gui
```

Comment:
Maybe instead of this, you could wrap the import in a try/except block in the normal spot and set a global `can_gui` flag that you then test here?